packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -163,8 +164,8 @@ class IOSDevice extends Device {
     // TODO(devoncarew): Handle startPaused, debugPort.
     printTrace('Building ${app.name} for $id');
 
-    // Step 1: Install the precompiled application if necessary.
-    bool buildResult = await buildIOSXcodeProject(app, buildForDevice: true);
+    // Step 1: Install the precompiled/DBC application if necessary.
+    bool buildResult = await buildIOSXcodeProject(app, mode, buildForDevice: true);
```

Comment:
I am not sure I understand the asymmetry. If I build the application in `debug` and and attempt to start it in `release`, the tools would still need to be able to understand the existing APK is invalid and must be rebuilt.